### PR TITLE
feat(app): add tooltip and disabled state to calibrate button in Desktop app

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add disabled state and tooltip to calibrate button when calibrationStatus-complete is false and switch the tooltip text by calibrationStatus reason.

close RAUT-555

Screenshot 2023-08-29 at 2 43 16 PM

# Test Plan

# Changelog

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupModules/SetupModulesList.tsx [hunk 1]
```diff
@@ -281,9 +293,22 @@ export function ModulesListItem({
...
     } else {
       return (
-        <TertiaryButton onClick={handleCalibrate}>
-          {t('calibrate_now')}
-        </TertiaryButton>
+        <>
+          <TertiaryButton
+            onClick={() => setShowModuleWizard(true)}
+            disabled={!calibrationStatus?.complete}
+            {...targetProps}
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupModules/SetupModulesList.tsx]: causing test error

File: app/src/organisms/Devices/ProtocolRun/SetupModules/SetupModulesList.tsx [hunk 2]
```diff
@@ -251,35 +265,44 @@ export function ModulesListItem({
...
-          status={moduleConnectionStatus}
-          backgroundColor={COLORS.successBackgroundLight}
-          iconColor={COLORS.successEnabled}
-          textColor={COLORS.successText}
-        />
-      )
-    } else {
-      return (
-        <TertiaryButton onClick={handleCalibrate}>
+  let renderModuleStatus: JSX.Element
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupModules/SetupModulesList.tsx]: it is a bit easier to read in my opinion to initialize the `renderModuleStatus` to = the else, then you can delete the else.

```suggestion
  let renderModuleStatus: JSX.Element = (
      <>
        <TertiaryButton
          {...targetProps}
          onClick={() => setShowModuleWizard(true)}
          disabled={!calibrationStatus?.complete}
        >
          {t('calibrate_now')}
        </TertiaryButton>
        {!calibrationStatus?.complete && calibrationStatus?.reason != null ? (
          <Tooltip tooltipProps={tooltipProps}>
            {calibrationStatus.reason === 'attach_pipette_failure_reason'
              ? t('attach_pipette_before_module_calibration')
              : t('calibrate_pipette_before_module_calibration')}
          </Tooltip>
        ) : null}
      </>
    )
```

File: app/src/organisms/Devices/ProtocolRun/SetupModules/__tests__/SetupModulesList.test.tsx [hunk 3]
```diff
@@ -233,13 +242,46 @@ describe('SetupModulesList', () => {
...
-    const { getByText } = render(props)
+    const { getByText, getByRole } = render(props)
     getByText('Thermocycler Module')
     getByText('Slot A1+B1')
-    getByText('Calibrate now').click()
+    getByRole('button', { name: 'Calibrate now' }).click()
     getByText('mock ModuleWizardFlows')
   })
 
+  it('should render disabled button when pipette and module are not calibrated', async () => {
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupModules/__tests__/SetupModulesList.test.tsx]: is the async needed?

**Review by jerader — APPROVED** [review]
lgtm, left a few clean up suggestions

File: app/src/organisms/Devices/ProtocolRun/SetupModules/__tests__/SetupModulesList.test.tsx (hunk 3 above)
Comment (koji) [reply app/src/organisms/Devices/ProtocolRun/SetupModules/__tests__/SetupModulesList.test.tsx]: oh good catch, it's not needed

**Pull request merged**